code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardEventObserver.kt

```diff
@@ -0,0 +1,184 @@
+                if (event.aepUi.getTemplate() is SmallImageTemplate) {
+                    Log.trace(MessagingConstants.LOG_TAG, SELF_TAG, "SmallImageUI click")
+                    handleSmallImageTracking(smallImageTemplate, event.action, MessagingEdgeEventType.INTERACT)
+                }
+            }
+        }
+    }
+
+    private fun handleSmallImageTracking(template: SmallImageTemplate, interaction: UIAction?, eventType: MessagingEdgeEventType) {
+        val contentCardSchemaData = ContentCardMapper.instance.getContentCardSchemaDataForPropositionId(template.id)
+        contentCardSchemaData?.track(interaction?.toString(), eventType)
+    }
+}
+
+class DefaultContentCardCallback : ContentCardCallback {
+    override fun onCardDisplayed(event: UIEvent.Display<*, *>): Boolean {
+        return true
+    }
+
+    override fun onCardClick(event: UIEvent.Interact<*, *>): Boolean {
+        return true
+    }
+
+    override fun onCardDismiss(event: UIEvent.Dismiss<*, *>): Boolean {
+        return true
+    }
+
+    override fun onButtonClick(event: UIEvent.Interact<*, *>): Boolean {
+        return true
+    }
+
+    override fun onDismissButtonClick(event: UIEvent.Interact<*, *>): Boolean {
+        return true
+    }
+}
+
+/**
+ * Interface to handle different callback events which can occur for a displayed content card.
+ */
+interface ContentCardCallback {
+
+    /**
+     * Callback to invoke when a content card is displayed.
+     *
+     * @param event The [UIEvent.Display] which occurred.
+     * @return true if tracking should be handled by the Messaging extension, false otherwise.
+     */
+    fun onCardDisplayed(event: UIEvent.Display<*, *>): Boolean
+
+    /**
+     * Callback to invoke when a content card is clicked.
+     *
+     * @param event The [UIEvent.Interact] which occurred.
+     * @return true if tracking should be handled by the Messaging extension, false otherwise.
+     */
+    fun onCardClick(event: UIEvent.Interact<*, *>): Boolean
+
+    /**
+     * Callback to invoke when a content card is dismissed.
+     *
+     * @param event The [UIEvent.Dismiss] which occurred.
+     * @return true if tracking should be handled by the Messaging extension, false otherwise.
+     */
+    fun onCardDismiss(event: UIEvent.Dismiss<*, *>): Boolean
+
+    /**
+     * Callback to invoke when a content card button is clicked.
+     *
+     * @param event The [UIEvent.Interact] which occurred.
+     * @return true if tracking should be handled by the Messaging extension, false otherwise.
+     */
+    fun onButtonClick(event: UIEvent.Interact<*, *>): Boolean
+
+    /**
+     * Callback to invoke when a content card dismiss button is clicked.
+     *
+     * @param event The [UIEvent.Interact] which occurred.
+     * @return true if tracking should be handled by the Messaging extension, false otherwise.
+     */
+    fun onDismissButtonClick(event: UIEvent.Interact<*, *>): Boolean
```

Comment:
Wondering if we should club all interactions to a generic `onInteract` since buttons is not mandatory. And it would consistent with iOS. Thoughts?

Reply:
Also do you think we need both `onDismissButtonClick` and `onCardDismiss`? Since there's no way customer can do the disqualification, can we just inform them about the dismiss so they can update the UI?

Reply:
this is also resolved as i've updated the SmallImageTemplateEventHandler to handle a generic `onInteractEvent`